Library/Homebrew/utils/github.rb

```diff
@@ -508,3 +505,32 @@ def api_errors
      GitHub::RateLimitExceededError, GitHub::Error, JSON::ParserError].freeze
   end
 end
+
+class GitHubArtifactDownloadStrategy < AbstractFileDownloadStrategy
```

Comment:
Can you check that being defined in this file does not make it accessible in formulae?

Reply:
It unfortunately is. How about this: change the `fetch artifact` function to just return the final artifact url, which is then passed into brew pr-pull and does all the fetching and unpacking, which means the custom strategy can live in that file.

Reply:
Sounds good 👍.